### ISSUE TITLE
Can't backup

### ISSUE THREAD

**Issue by ignamartinoli** [spicetify/spicetify-cli#1732]
### 🔍 Have you checked Spicetify.app page for your issue?

- [X] I have checked the FAQ

### 🔍 Is there already an issue for your problem?

- [X] I have checked older issues, open and closed

### ℹ Environment / Computer Info

```Markdown
- Spotify version: 1.1.84.716.gc5f8b819
- Spicetify version: 2.10.1
```

### 📝 Description

1) I installed **Spotify** from zero.
2) I installed **Spicetify** from zero.
3) I changed this permissions
```shell
sicro@sicro ~ $ doas chmod a+wr /opt/spotify
sicro@sicro ~ $ doas chmod a+wr /opt/spotify/Apps -R
```
4) I edited `~/.config/spicetify/config-xpui.ini` with
```ini
prefs_path              = /home/sicro/.config/spotify/prefs
spotify_path            = /opt/spotify
```
5) I ran this command
```shell
sicro@sicro ~ $ spicetify backup
spicetify v2.10.1
Backing up app files:
error Cannot backup app files. Reinstall Spotify and try again.
```

**Comment by afonsojramos** [comment]
Have you tried deleting `.spicetify` and `.config/spicetify` and doing the process again?

**Comment by ignamartinoli** [comment]
I deleted `.spicetify` only. I'm gonna try deleting both right now and check if anything changes

**Comment by ignamartinoli** [comment]
Just tried. Removed **Spotify**, `~/.spicetify` and `~/.config/spicetify`. Reinstalled Spotify and Spicetify. Updated **Spotify**'s permissions. Ran `spicetify backup` and got
```shell
sicro@sicro ~ $ spicetify backup
error Could not detect Spotify location.
success Default config-xpui.ini generated.
error Cannot detect Spotify location. Please manually set "spotify_path" in config-xpui
```
so I edited `~/.config/spicetify/config-xpui.ini` to
```ini
spotify_path            = /opt/spotify
```
then I retried and got the message
```shell
sicro@sicro ~ $ spicetify backup
spicetify v2.10.1
Backing up app files:
error Cannot backup app files. Reinstall Spotify and try again.
```

**Comment by wael444** [comment]
do you have user set permissions on `/opt/spotify`?

**Comment by ignamartinoli** [comment]
I do
```shell
sicro@sicro ~ $ ls -l /opt/
drwxrwxrwx  4 root root 4096 May 29 20:38 spotify

sicro@sicro ~ $ ls -l /opt//spotify
drwxrw-rw- 5 sicro sicro 4096 May 10 13:27 Apps

sicro@sicro ~ $ ls -l /opt/spotify/Apps
drwxrw-rw- 5 sicro sicro 4096 May 10 13:27 login
drwxrw-rw- 7 sicro sicro 4096 May 10 18:43 xpui
drwxrw-rw- 2 sicro sicro 4096 May 10 13:27 zlink
```

**Comment by ignamartinoli** [comment]
@ririxidev a couple of times. One thing that caught my eye is that the folder tree of **Spotify** in **Gentoo** is

```
/opt/spotify/
├── Apps/
└── spotify-client/
```

I am not sure if that `spotify-client` directory is present in the others distros.

I noticed that there is [this](https://gpo.zugaina.org/Overlays/src_prepare-overlay/app-misc/spicetify-cli) overlay for **Spicetify**. If you click on "View" it shows the details of the ebuild. At the bottom it suggests

```ebuild
pkg_postinst() {
	elog "Spicetify requires a Spotify install that it can modify. Version 1.1.72 and 1.1.73"
	elog "are supported by this version of spicetify. To be able to modify system installed "
	elog "Spotify like media-sound/spotify then you have to allow spicetify to read and write"
	elog "to its install location like shown below."
	elog "# chmod a+wr /opt/spotify/spotify-client"
	elog "# chmod a+wr /opt/spotify/Apps -R"
	elog ""
	elog "Otherwise you can install spotify to a user modifiable location like as a flatpak:"
	elog "https://github.com/khanhas/spicetify-cli/wiki/Installation#spotify-installed-from-flatpak"
}
```

In [this](https://spicetify.app/docs/advanced-usage/installation#spotify-installed-from-aur) section of the official **Spicetify** documentation, it is suggested to change the permissions as

```bash
sudo chmod a+wr /opt/spotify
sudo chmod a+wr /opt/spotify/Apps -R
```

Which conflicts on how it is suggested on that **Spicetify** ebuild.

This is all I could find out in the meantime

**Comment by ignamartinoli** [comment]
@ririxichan yea, I forgot to mention. When I realized that I ran `chmod` to `spotify-client` but that wasn't enough to make it work. Maybe there is some internal code that has to be modified to make it run on Gentoo

**Comment by ignamartinoli** [comment]
Hey, I found that making this work on **Gentoo** should not be particularly difficult.
I made a better dig on that Overlay (the term used in **Gentoo** for an unofficial package repository).
If you check [this](https://github.com/gentoo-mirror/src_prepare-overlay/blob/master/app-misc/spicetify-cli/files/add_gentoo_install_dir.patch) file it shows that is only needed to add this directory to the potential list

```
 	potentialList := []string{
 		"/opt/spotify/",
+		"/opt/spotify/spotify-client/",
 		"/usr/share/spotify/",
 		"/var/lib/flatpak/app/com.spotify.Client/x86_64/stable/active/files/extra/share/spotify/",
 	}
```

and on that same file there is another small directory naming change that I think is probably mostly optional.
I'm not too familiar with **Git** so I don't really know how to do a pull/push request (I'm really dying of the shame right now for not knowing that) but I can give anybody assistance on testing it before merging to `master`
